python-lib/config.py

```diff
@@ -84,23 +90,21 @@ def get_hdfs_location(dataset_config: Mapping[AnyStr, Any], sf_stage_name: AnySt
 
 def get_table_schema_sql(sf_table_name: AnyStr) -> AnyStr:
     """
-    Generates SQL to extract table columns from DB
-    :param sf_table_name: Snowflake table in format of "schema"."table". Note that schema and table are case-sensitive.
+    Generates SQL to extract a table's column schema
+    :param sf_table_name: Snowflake table
     :return: SQL for getting table columns
     """
-    tab = sf_table_name.replace('"', '')
-    schema = tab.split('.')[0]
-    table = tab.split('.')[1]
 
-    sql = f'''
-SELECT column_name AS "name", data_type AS "originalType"
-FROM information_schema.columns
-WHERE table_name = '{table}'
-'''
-    if schema:
-        sql += f" AND table_schema = '{schema}'"
+    tbl_sch_re = re.search(r'("?(?P<schema>[^"]+)"?\.)?"?(?P<table>[^"]+)"?', sf_table_name)
```

Comment:
Regular expressions aren't easy to read, sure. But that's the nature of regular expressions.

The original code didn't work when the input was just `"TABLE_NAME"` without also writing additional checks.

Reply:
If you want to use Regexp this is fine, but in this case the code needs to document what this expression is doing.
You can put what table formats this will handle for example.

Reply:
I've added comments to describe the regexp. Examples of what it will handle are self documenting in the associated tests (which I've also added additional commentary to).